include/albatross/src/utils/compress.hpp

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (C) 2023 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef ALBATROSS_UTILS_COMPRESS_H
+#define ALBATROSS_UTILS_COMPRESS_H
+
+namespace albatross {
+
+namespace zstd {
+
+namespace internal {
+
+inline std::string compress(const void *data, std::size_t size_bytes,
+                            int compression_level) {
+  const std::size_t compressed_bound = ZSTD_compressBound(size_bytes);
+
+  std::string compressed{};
+  compressed.resize(compressed_bound);
+
+  const std::size_t compressed_size =
+      // I solemnly swear that I am up to no good.
```

Comment:
Maybe describe the situation you think could cause a failure here and/or why it won't happen, in place of this and the below comments.

Reply:
Lol, I'll remove those.